robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,216 @@
+import logging
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents Report Portal in ReportPortal
+
+    This holds the properties and function to interact with the Report Portal
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal to be used as a base in all child classes
+        :returns Base url for API request
+        """
+        _api_url = f'{self.rp_url}/api/v1/{self.rp_project}/'
+        return _api_url
```

Comment:
Consider just one-lining this property and not naming the string. A lot of these properties can be cached as well, as I don't think project/url/api_key will change while requesting launches. I'd _almost_ suggest creating a singleton instance in this module for import, but I don't think it'll really save us anything.

Reply:
Yeah, hence I didn't think of caching them not going to save much time.